types/types.go

```diff
@@ -89,6 +90,22 @@ func (o *NullBool) UnmarshalBSONValue(t bsontype.Type, bytes []byte) error {
 
 	return nil
 }
+func (o *NullBool) JSONUnmarshal(data []byte) error {
+	if len(data) <= 0 {
+		o.Valid = false
+		o.Bool = false
+		return nil
+	}
+
+	var isTrue bool
```

Comment:
for simple types such as JSON `boolean`, there are only two possible values, we can make a happy path

```go

switch(utils.UnsafeString(data) {
case "true":
case "false" 
default:
 // Fall back to JSON marshaler to the thing
```